File: 6_kyu/Stop_gninnipS_My_sdroW.lua
local solution = {}

-- Reverses all words with 5 or letters,
-- Argument:
--    sentence - String of word separated by space.
-- Return:
--    Copy of the string, all 5 or more letter words reversed.
function solution.spin_words(sentence)
  if type(sentence) == "string" then
    return sentence:gsub("%S%S%S%S%S+",string.reverse)
  else
    -- ignores invalid argument
    return ""
  end
end

return solution

File: 6_kyu/Stop_gninnipS_My_sdroW.py
import re

# Reverses all words with 5 or more letters in the argument sentence.
# Supports only ASCII letters a-z and A-Z.
def spin_words(sentence: str) -> str:
    return WORD_5CHAR.sub(lambda m:m.group(0)[::-1], sentence)

# Pattern of word 5 or more letters
WORD_5CHAR = re.compile(r"\b[a-zA-Z]{5,}\b")